### ISSUE TITLE
Not all Highlights Write to File

### ISSUE THREAD

**Issue by rmuser2** [koreader/koreader#8062]
* KOReader version: 2021.07
* Device: Remarkable 2

#### Issue
I have been using KOReader to read a lot of PDF books and articles, I use Goodsync to sync these files to my Google Drive.
My highlights have not been hard written to the PDFs as I intended, and set KOReader to do. 

Since noticing that there are no highlights on the cloud copy of the PDF, I have highlighted sections as a test and the test highlights are being written to the file and are transferring to the cloud, but all the previous highlights do not transfer, despite being visible when I read the PDF on KOReader.

I can't seem to find a way to make KOReader refresh the PDF or write everything to the file.

**Comment by mergen3107** [comment]
Try this setting: Top menu - Gear - Document - Save document (write highlights into PDF). 
Try to switch it to Prompt and see if it is properly asking you about this. Also note, when it asks. Then check whether it actually wrote the highlights into metadata.lua file in the respective .sdr folder (it is in the same location as book, starts with the same filename). 

I suspect it should write on close, on document switching or something like that. Not sure if it is related to Auto-save book metadata option above.

**Comment by rmuser2** [comment]
I have since switched it to prompt, and now the highlights are being written consistently. 

I now have an issue, as per a new issue post, relating to saved/written highlights still having the metadata highlight applied to a paragraph when it is opened next.

**Comment by poire-z** [comment]
Do you mean that it's just the highlights made *before* you enabled "Save highlights into PDF" that are not written into the PDF that are missing ? (in which case, it's understandable: we write them only when you make them - nothing there to batch write into the PDF the old koreader-only highlights) ?

Or that you still get missing ones even after you enabled "Save highlights into PDF" and prompt/always have different behaviours?

**Comment by rmuser2** [comment]
Thanks for the reply and time. 

Consider a PDF with two sentences. You highlight sentence one with KOReader, save and write to the PDF, sync the RM and then view that PDF on your computer. The highlight is there and Acrobat can delete it if needed = great!

Now when you open that PDF in KOReader a second time, the highlight which was written to the PDF is there (of course) in addition to a metadata highlight over the same sentence. In essence, KOReader does not remove the metadata highlight after it has written it to the PDF. 

As I scan through a PDF book on KOReader, I have darker highlights where the written highlight and the meta highlight overlap. Saving the PDF again does not seem to apply the metadata highlight a second time which is something.

Another issue is that the highlighter is just lightening the text and this makes the test go grey, which I can not read.

I think the RM2 will be up for sale next week! I had hoped KOReader was going to be its saving grace.

**Comment by rmuser2** [comment]
Yes, the former seems to be it. I could have highlighted the whole book, and only switched on the setting on the last page. This results in only the last page having highlighted text.
